### PR TITLE
refactor portfolio menu

### PR DESCRIPTION
**Pull request by jmaslek**
This PR replaces the port_menu with the port_controller as per the previous pr conversation.  also updated requirements.txt with robin-stocks and alpaca trade api (next project)

@DidierRLopes on my end, the linting failed due to the use of eval - so I am not sure what to replace with (also a 'pointless string' which is in the menu?)

edit -- apparently the listing passed!

### CODE REVIEW

File: gamestonk_terminal/portfolio/port_controller.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    ]
+
+    def __init__(self):
+        self.port_parser = argparse.ArgumentParser(add_help=False, prog="port")
+        self.port_parser.add_argument("cmd", choices=self.CHOICES)
+        self.brokers_list = set()
+
+    @staticmethod
+    def print_help(broker_list):
+        if not broker_list:
```
Comment (DidierRLopes) [inline gamestonk_terminal/portfolio/port_controller.py:34]: You don't need these 7 lines, look at my next comment

File: gamestonk_terminal/portfolio/port_controller.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        """ Print help """
+        print("\nPortfolio:")
+        print("   help          show this menu again")
+        print(
+            "   q             quit this menu, and shows back to main menu, logs out of brokers"
+        )
+        print("   quit          quit to abandon program, logs out of brokers")
+        print("   login         login to your brokers")
+        print("\nCurrent Brokers :" + print_broke)
```
Comment (DidierRLopes) [inline gamestonk_terminal/portfolio/port_controller.py:49]: You can do something like this in one line:
```
print(f"\nCurrent Brokers : {('', ', '.join(broker_list))[bool(broker_list)]}")
```

So, `', '.join(['a', 'b', 'c']) ` would resulting in the string `a, b, c`. 

Then you use the ternary operation
```
('A', 'B')[False] = 'A'
('A', 'B')[True] = 'B'
```

File: gamestonk_terminal/portfolio/port_controller.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            rh_api.show_holdings()
+        except Exception as e:
+            print(e)
+            print("")
+
+
+def menu():
+    """Portfolio Analysis Menu"""
+
+    port_controller = PortfolioController()
```
Comment (DidierRLopes) [inline gamestonk_terminal/portfolio/port_controller.py:137]: To follow the other menus:
```
fundamental analysis
technical analysis
comparison analysis
behavioural analysis
..
```

should this be `pa` for `portfolio analysis` ?

**Review by DidierRLopes — COMMENTED** [review]
Just commented on a code improvement.

All the rest looks good to go! Good work!!

File: gamestonk_terminal/portfolio/port_controller.py (hunk 1 above)
Comment (jmaslek) [reply gamestonk_terminal/portfolio/port_controller.py:34]: I assume you mean starting at `if not broker_list` ?  It should be a static method, right?

File: gamestonk_terminal/portfolio/port_controller.py (hunk 2 above)
Comment (jmaslek) [reply gamestonk_terminal/portfolio/port_controller.py:49]: Love the python golf!

File: gamestonk_terminal/portfolio/port_controller.py (hunk 1 above)
Comment (DidierRLopes) [reply gamestonk_terminal/portfolio/port_controller.py:34]: Yea, I mean that the line of code that I mentioned below can do this all in 1 go :p

File: gamestonk_terminal/portfolio/port_controller.py (hunk 2 above)
Comment (DidierRLopes) [reply gamestonk_terminal/portfolio/port_controller.py:49]: You live and you learn :D

**Comment by jmaslek** [comment]
made those changes.  Still added the if not brokers_list to print None when not logged in

**Comment by DidierRLopes** [comment]
> made those changes. Still added the if not brokers_list to print None when not logged in

Nicee!! I'll merge this once checks are done

You could also have added the **None** in that line, here:
```
print(f"\nCurrent Brokers : {('None', ', '.join(broker_list))[bool(broker_list)]}")
```

**Pull request merged**